map/challenges-07.test.js

```diff
@@ -0,0 +1,289 @@
+'use strict';
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 1
+//
+// Write a function named forLoopTwoToThe that, given an array of integers as input, iterates over
+// the array and returns a new array. The returned array should containing the result of raising 2
+// to the power of the original input element.
+//
+// You may choose to complete this challenge using a for loop, for...in syntax, or for...of syntax.
+//
+// For example, twoToThe([1,2,3]) returns [2,4,8] because 2 ^ 1 = 2, 2 ^ 2 = 4, and 2 ^ 4 = 8.
+// ------------------------------------------------------------------------------------------------
+
+const forLoopTwoToThe = (input) => {
+  let newArr = [];
+  
+  for (let i = 0; i < input.length; i++) {
+    newArr.push(Math.pow(2, input[i]));
+  }
+
+  return newArr;
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 2
+//
+// Write a function named forEachTwoToThe that produces the same output as your forLoopTwoToThe
+// function from challenge 1, but uses forEach instead of a for loop.
+// ------------------------------------------------------------------------------------------------
+
+const forEachTwoToThe = (input) => {
+
+  let newArr = [];
+
+  input.forEach( value => {
+    newArr.push(Math.pow(2, value));
+  });
+
+  return newArr;
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 3
+//
+// Write a function named mapTwoToThe that produces the same output as your forLoopTwoToThe function
+// from challenge 1 and your forEachTwoToThe function from challenge 2, but uses map
+// instead of a for loop or forEach.
+// ------------------------------------------------------------------------------------------------
+
+const mapTwoToThe = (input) => {
+  let newArr = input.map(value => Math.pow(2, value));
+  return newArr;
+};

```

Comment:
making a new variable is unnecessary, simply `return input.map(num => Math.pow(2, num));`